### PR TITLE
Transfer part of project Data access to use a Logger

### LINKED ISSUE #451: Transfer part of project Data access to use a Logger

### PR DESCRIPTION
**Pull request by ivanboika**
Task:
At the moment, the project has a utility for logging. Instead, we use output to standard error.
Do not do like this.

### CODE REVIEW

File: DataAccess/DataAccess.Postgre/CMakeLists.txt [hunk 1]
```diff
@@ -20,3 +20,6 @@ set_target_properties(${TARGET} PROPERTIES FOLDER ${FOLDER})
 target_link_libraries(${TARGET}
         PUBLIC PostgreSQL::pq libpqxx::libpqxx
         PUBLIC Utility.Public DataAccess.Public Network.Public)
+        PUBLIC CONAN_PKG::libpq CONAN_PKG::libpqxx
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/CMakeLists.txt]: Rebase needed

**Review by memorisecodead — APPROVED** [review]

**Review by memorisecodead — APPROVED** [review]

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 2]
```diff
@@ -32,6 +35,12 @@ Utility::ChannelLeaveCodes ChannelsRepository::leaveChannel(const Network::Chann
     pTable->changeTable("user_channels");
     if (!findIdChannel.has_value())
     {
+        Base::Logger::FileLogger::getInstance().log(
+            '[' + "channel id: " + 
+            std::to_string(findIdChannel.value()[0][0].as<std::uint64_t>()) +
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: If the channel is not found, how to get it from the database?

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 3]
```diff
@@ -47,13 +56,31 @@ Utility::ChannelLeaveCodes ChannelsRepository::leaveChannel(const Network::Chann
...
+                std::to_string(findChannel.value()[0][0].as<std::uint64_t>()) + " failed\n",
+                Base::Logger::LogLevel::ERR);
+
             return Utility::ChannelLeaveCodes::FAILED;
         }
     }
     else
     {
+        Base::Logger::FileLogger::getInstance().log(
+            '[' + "channel id: " + std::to_string(findChannel.value()[0][1].as<std::uint64_t>()) +
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: If the channel is not found, how to get it from the database?

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 4]
```diff
@@ -63,13 +90,25 @@ Utility::ChannelDeleteCode ChannelsRepository::deleteChannel(const Network::Chan
     auto findChannel = pTable->Select()->columns({"creator_id, id"})->Where("channel_name = '" + channel.channelName + "'")->execute();
     if (!findChannel.has_value())
     {
+        Base::Logger::FileLogger::getInstance().log(
+            '[' + "channel id: " + 
+            std::to_string(findChannel.value()[0][1].as<std::uint64_t>()) +
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: The same...

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 5]
```diff
@@ -63,13 +90,25 @@ Utility::ChannelDeleteCode ChannelsRepository::deleteChannel(const Network::Chan
...
         return Utility::ChannelDeleteCode::CHANNEL_NOT_FOUND;
     }
 
     auto creatorlID  = findChannel.value()[0][0].as<uint64_t>();
     auto channelID   = findChannel.value()[0][1].as<uint64_t>();
     if (creatorlID != channel.creatorID)
     {
+        Base::Logger::FileLogger::getInstance().log(
+            '['+ "channel id: " + std::to_string(channelID) + 
+            "] User id " + std::to_string(creatorlID) +
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp:103]: Looks confusing 
[channel id 213123] User id 666 is not the creator of channel id 1231321
better would be:
[channel id 213123] User id 666 is not the creator of this channel

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 6]
```diff
@@ -193,6 +270,10 @@ std::vector<Network::MessageInfo> MessagesRepository::getMessageHistory(const st
             result.emplace_back(mi);
         }
     }
+     Base::Logger::FileLogger::getInstance().log(
+        '[' + "channel id: " + std::to_string(channelID) +
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp:269]: Could be flooded, need testing

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 7]
```diff
@@ -277,11 +379,21 @@ Utility::EditingMessageCodes MessagesRepository::editMessage(const Network::Mess
                                 ->execute();
     if (!isPresentInTable.has_value())
     {
+        Base::Logger::FileLogger::getInstance().log(
+            '[' +"channel id: "+ std::to_string(mi.channelID) +']' +
+            " Editing message failed\n",
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Not clear what happend

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 8]
```diff
@@ -319,7 +445,10 @@ Utility::ReactionMessageCodes MessagesRepository::updateMessageReactions(const N
 
     if (reactionInfo == mi.reactions.end())
     {
-        std::cerr << "Invalid reaction info supplied" << std::endl;
+        Base::Logger::FileLogger::getInstance().log(
+            "Invalid reaction info supplied",
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Not clear what meaning

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 2 above)
Comment (Orange-hanter) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: This should be changed

**Review by Orange-hanter — CHANGES_REQUESTED** [review]

File: DataAccess/DataAccess.Postgre/CMakeLists.txt (hunk 1 above)
Comment (ivanboika) [reply DataAccess/DataAccess.Postgre/CMakeLists.txt]: Already, or I'm mistaken

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 2 above)
Comment (ivanboika) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Done

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 3 above)
Comment (ivanboika) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Fixed

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 4 above)
Comment (ivanboika) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Fixed

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 5 above)
Comment (ivanboika) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp:103]: Done

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 9]
```diff
@@ -47,13 +54,30 @@ Utility::ChannelLeaveCodes ChannelsRepository::leaveChannel(const Network::Chann
                           ->execute();
         if (result.has_value())
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp:55]: There needs to be a reverse logic here, right?

Comment (Orange-hanter) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp:55]: What should actually be returned  by the delete operation?

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 10]
```diff
@@ -32,6 +35,10 @@ Utility::ChannelLeaveCodes ChannelsRepository::leaveChannel(const Network::Chann
     pTable->changeTable("user_channels");
     if (!findIdChannel.has_value())
     {
+        Base::Logger::FileLogger::getInstance().log(
+            "Leaving from channel failed because channel was not found\n ",
+            Base::Logger::LogLevel::WARNING);
```
Comment (Orange-hanter) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Error

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 7 above)
Comment (ivanboika) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Fixed

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 8 above)
Comment (ivanboika) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: Fixed

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp (hunk 10 above)
Comment (Orange-hanter) [reply DataAccess/DataAccess.Postgre/PostgreRepositories.cpp]: done

**Review by Orange-hanter — APPROVED** [review]

**Pull request merged**